src/main/kotlin/uk/gov/justice/digital/assessments/restclient/AssessmentUpdateRestClient.kt

```diff
@@ -18,12 +25,64 @@ class AssessmentUpdateRestClient {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
-  fun createOasysOffender(crn: String, user: String = "STUARTWHITLAM", area: String = "WWS", deliusEvent: Int = 123456): Long? {
+  fun createOasysOffender(
+    crn: String,
+    user: String = "STUARTWHITLAM",
+    area: String = "WWS",
+    deliusEvent: Int = 123456
+  ): Long? {
     log.info("Creating offender in OASys for crn: $crn, area: $area, user: $user, delius event: $deliusEvent")
     return webClient
       .post("/offenders", CreateOffenderDto(crn, area, user, deliusEvent))
       .retrieve()
+      .onStatus(
+        { it == HttpStatus.FORBIDDEN },
+        {
+          log.error("User $user does not have permission to create offender with crn $crn")
+          it.bodyToMono(ErrorResponse::class.java)
+            .flatMap { error -> Mono.error(UserNotAuthorisedException(error.developerMessage)) }
+        }
+      )
+      .onStatus(
+        { it == HttpStatus.CONFLICT },
+        {
+          log.error("Duplicate records found for $crn")
+          it.bodyToMono(ErrorResponse::class.java)
```

Comment:
I don't like using flatMap here but not sure how else to get the ErrorResponse out without using block() which doesn't work

Reply:
Then it's the right thing to do :)

It feels like there's another little helper lambda in here if you were feeling keen

fun <R> handleError(errorMessage): something -> Nothing
{
  return (resp) -> {
    log.error(errorMessage)
    resp.bodyToMono(ErrorResponse::class.java)
            .flatMap { error -> Mono.error(UserNotAuthorisedException(error.developerMessage)) 
    }
 }